src/cswinrt/code_writers.h

```diff
@@ -5257,7 +5257,7 @@ try
 }
 catch (Exception __exception__)
 {
-global::WinRT.ExceptionHelpers.SetErrorInfo(__exception__);
```

Comment:
Is the changes in this file basically calling ReportUnhandledError in all our Do_ABI functions?  If that is right, that would mean we are saying that any error crossing the ABI boundary is an unhandled error.  Is that what we want or did we want it such that any errors throw in handlers are reported as unhandled errors.

Reply:
Yes, this is reporting all unhandled errors (uncaught managed exceptions) at the ABI and passing them along to RoReportUnhandledError, which is what the WRL wrappers do, and what the UWP behavior is.

Reply:
With UWP, Windows.UI.Xaml.dll relied on twinapi.appcore.dll's CoreApplication implementation, which routed all app events thru WRL's event implementation.  WRL's InvokeDelegates in turn would catch non-zero HRESULTs and invoke RoReportFailedDelegate.  

But with WinAppSDK, Microsoft.UI.Xaml.dll invokes marshaled event handlers directly and just propagates non-zero HRESULTs up the stack to Application::Start (where the debugger then presents an unhelpful 0xc000027b message) or simply ignores them (e.g., CFTMEventSource::Raise). 

We could change Microsoft.UI.Xaml.dll to try and catch all callback/event invocation non-zero HRESULTs and pass them to RoReportUnhandledError (or RoReportFailedDelegate), but we'd still want to call SetErrorInfo to get the language exception (stack, message, etc).  That being the case, it's easiest to code-gen that call alongside SetErrorInfo.

Reply:
@manodasanW I've removed all error reporting changes, and implemented them in Microsoft.UI.Xaml.dll instead, where they apply to both C# and C++ apps.